Store service #30 - Upload/Download Process File

amarbajric opened the pull request:
WIP

**DONE**
- Implemented endpoints and logic to save (i.e. upload) and download a file to a specific process

**Upload File to process:**
- Send a `POST` to `/api/store/process/<processId>/uploadProcessFile`
  - `Content-Type` in Header should be `multipart/form-data`
  - set the actual `file` as value under the key `file` in the body
- File is saved under a column `process_file` in the `processstore` table (i.e. longblob)

**Get File of a process:**
- Send a `GET` to `/api/store/process/<processId>/getProcessFile`
  - Handle response with Angular and save the data stream into a file in the frontend and save the file as `*.owl`

stfnltnr committed:
added endpoints to process-store, for detailed information check API documentation and pull requests

Changes to be committed:
        modified:   ../Gateway/src/main/java/at/fhjoanneum/ippr/gateway/api/controller/ProcessStoreGatewayController.java
        modified:   ../Gateway/src/main/java/at/fhjoanneum/ippr/gateway/api/controller/user/HttpHeaderUser.java
	modified:   ../Gateway/src/main/java/at/fhjoanneum/ippr/gateway/api/services/impl/ProcessStoreCallerImpl.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/controller/ProcessStoreController.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreBuilder.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreObjectImpl.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/repositories/ProcessStore.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/services/ProcessStoreService.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/services/ProcessStoreServiceImpl.java
	modified:   ../SpringCommons/src/main/java/at/fhjoanneum/ippr/commons/dto/processstore/ProcessStoreDTO.java

stfnltnr committed:
gradle.bat and gradle-wrapper.properties for W10 startup

Commit:
Merge branch 'dev' into store-service-#30

stfnltnr committed:
added Rating and ApprovalMessage

stfnltnr committed:
Merge branch 'dev' into store-service-#30

Commit:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

Commit:
Added endpoint where a file can be uploaded to a process

Commit:
Added endpoint to download processFile

Commit:
added missing files

Commit:
Merge branch 'dev' into store-service-#30

Commit:
added changelog entry

stfnltnr committed:
added endpoints for mapping processes to orgaIds (BUYING)

Commit:
removed DELETE_FINAL folder

stfnltnr committed:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

stfnltnr commented:
Added ProcessOrgaMapping for buying a process with the provided processId via URL

```POST localhost:10000/api/store/process/{processId}/buy```

`orgaId` & `userId` is provided via body
```
{
	"orgaId":"2",
	"userId":"1"
}
```
Added endpoint to retrieve purchased processes according to processOrgaMapping
``` GET localhost:10000/api/store/processes/byOrga/{orgaId}``` lists all bought processes for the given `orgaId`

Example of my test data

id | orga_id | process_id | user_id
-- | -- | -- | --
1 | 1 | 3 | 2
2 | 1 | 2 | 2
3 | 2 | 2 | 1

Receive purchased processes ```GET localhost:10000/api/store/processes/byOrga/2```
```
[
    {
        "processId": 2,
        "processName": "name2",
        "processDescription": "testasdas asda as asd as ad asd ad asd ad asd dd",
        "processCreator": "t1231est",
        "processCreatedAt": 1592668800000,
        "processVersion": 2,
        "processPrice": 222.222,
        "processApprover": null,
        "processApproverComment": null,
        "processApprovedDate": null,
        "approved": null
    }
]
```

stfnltnr committed:
fixed issue with approved: null / false
Changes to be committed:
	modified:   ProcessStore/src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreBuilder.java
	modified:   ProcessStore/src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreObjectImpl.java
	modified:   ProcessStore/src/main/java/at/fhjoanneum/ippr/processstore/repositories/ProcessStore.java
        modified:   ProcessStore/src/main/resources/application.properties
	modified:   SpringCommons/src/main/java/at/fhjoanneum/ippr/commons/dto/processstore/ProcessStoreDTO.java

Commit:
Added properties for max file upload size which is now set to 10MB.

Commit:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30
